src/pages/playground/Playground.tsx

```diff
@@ -230,15 +236,103 @@ export async function handleHash(
+      await overwriteFilesInWorkspace(workspaceLocation, fileSystem, files);
+    }
+
+    // BrowserFS does not provide a way of listening to changes in the file system, which makes
+    // updating the file system view troublesome. To force the file system view to re-render
+    // (and thus display the updated file system), we first disable Folder mode.
+    dispatch(setFolderMode(workspaceLocation, false));
+    const isFolderModeEnabled = convertParamToBoolean(configObj.isFolder?.toString()) ?? false;
+
+    // If Folder mode should be enabled, enabling it after disabling it earlier will cause the
+    // newly-added files to be shown. Note that this has to take place after the files are
+    // already added to the file system.
+    dispatch(setFolderMode(workspaceLocation, isFolderModeEnabled));
+
+    // By default, open a single editor tab containing the default playground file.
+    const editorTabFilePaths = configObj.tabs
+      ?.split(',')
+      .map(decompressFromEncodedURIComponent) ?? [defaultFilePath];
+
+    // Remove all editor tabs before populating with the ones from the query string.
+    dispatch(
+      removeEditorTabsForDirectory(workspaceLocation, WORKSPACE_BASE_PATHS[workspaceLocation])
+    );
+    // Add editor tabs from the query string.
+    editorTabFilePaths.forEach(filePath =>
+      // Fall back on the empty string if the file contents do not exist.
+      dispatch(addEditorTab(workspaceLocation, filePath, files[filePath] ?? ''))
+    );
+
+    // By default, use the first editor tab.
+    const activeEditorTabIndex = convertParamToInt(configObj.tabIdx?.toString()) ?? 0;
+    dispatch(updateActiveEditorTabIndex(workspaceLocation, activeEditorTabIndex));
+    if (chapter) {
+      // TODO: To migrate the state logic away from playgroundSourceChapter
+      //       and playgroundSourceVariant into the language config instead
+      const languageConfig = getLanguageConfig(chapter, configObj.variant as Variant);
+      handlers.handleChapterSelect(chapter, languageConfig.variant);
+      // Hardcoded for Playground only for now, while we await workspace refactoring
+      // to decouple the SicpWorkspace from the Playground.
+      dispatch(playgroundConfigLanguage(languageConfig));
+    }
+
+    const execTime = Math.max(
+      convertParamToInt(configObj.exec?.toString() || '1000') || 1000,
+      1000
+    );
+    if (execTime) {
+      handlers.handleChangeExecTime(execTime);
+    }
+  }
+}
+
 const Playground: React.FC<PlaygroundProps> = props => {
   const { isSicpEditor } = props;
   const workspaceLocation: WorkspaceLocation = isSicpEditor ? 'sicp' : 'playground';
   const { isMobileBreakpoint } = useResponsive();
 
   const [deviceSecret, setDeviceSecret] = useState<string | undefined>();
-  const location = useLocation();
   const navigate = useNavigate();
   const store = useStore<OverallState>();
+  const location = useLocation();
   const searchParams = new URLSearchParams(location.search);
```

Comment:
Can we undo this line reordering to keep the PR clean?

Reply:
Moved this line back to line333.